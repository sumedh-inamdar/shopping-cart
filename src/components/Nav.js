import React from 'react';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPiedPiperSquare } from '@fortawesome/free-brands-svg-icons';
import { faShoppingCart } from '@fortawesome/free-solid-svg-icons';
// import logo from '../assets/logo.jpg';

function toggleMobileMenu() {
  const menu = document.querySelector('.mobile-menu');
  menu.classList.toggle('hidden');
}

export default function Nav() {
  return (
    <nav className="bg-gradient-to-r from-teal-500 to-indigo-500 text-white">
      <div className="max-w-6xl mx-auto px-4">
        <div className="flex justify-between">
          <div className="md:hidden flex items-center">
            <button
              className="outline-none mobile-menu-button"
              onClick={toggleMobileMenu}>
              <svg
                className="w-10 h-10"
                x-show="!showMenu"
                fill="none"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                stroke="currentColor">
                <path d="M4 6h16M4 12h16M4 18h16"></path>
              </svg>
            </button>
          </div>
          <div className="flex space-x-7 min-w-fit">
            <div>
              <Link
                className="flex items-center text-3xl md:text-4xl py-4 px-2"
                to="/">
                <FontAwesomeIcon
                  icon={faPiedPiperSquare}
                  className="mr-2 pb-2.5"
                />
                <div className="flex flex-row justify-start content-center">
                  <div className="font-bold">TENNS</div>
                  <div className="font-light">CENTRL</div>
                </div>
              </Link>
            </div>
            <div className="hidden md:flex items-center space-x-1">
              <Link
                className="py-4 px-2 navLink text-2xl font-light hover:text-zinc-600 transition duration-300"
                to="/">
                HOME
              </Link>
              <Link
                className="py-4 px-2 navLink text-2xl font-light hover:text-zinc-600 transition duration-300"
                to="/store">
                STORE
              </Link>
              <Link
                className="py-4 px-2 navLink text-2xl font-light hover:text-zinc-600 transition duration-300"
                to="/about">
                ABOUT
              </Link>
            </div>
          </div>
          <div className="flex items-center space-x-3 ">
            <Link
              className="text-2xl py-2 px-2 hover:text-zinc-600 transition duration-300"
              to="/checkout">
              <FontAwesomeIcon icon={faShoppingCart} />
            </Link>
          </div>
        </div>
      </div>
      <div className="hidden mobile-menu">
        <ul className="">
          <li className="active">
            <Link
              to="/"
              className="block text-sm px-2 py-4 text-white bg-green-600 font-semibold">
              HOME
            </Link>
          </li>
          <li>
            <Link
              to="/store"
              className="block text-sm px-2 py-4 hover:bg-green-600 transition duration-300">
              STORE
            </Link>
          </li>
          <li>
            <Link
              to="/about"
              className="block text-sm px-2 py-4 hover:bg-green-600 transition duration-300">
              ABOUT
            </Link>
          </li>
        </ul>
      </div>
    </nav>
  );
}
