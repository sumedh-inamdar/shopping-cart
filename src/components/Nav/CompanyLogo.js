import React from 'react';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPiedPiperSquare } from '@fortawesome/free-brands-svg-icons';
import PropTypes from 'prop-types';

export default function CompanyLogo({ clickHandler }) {
  return (
    <div onClick={clickHandler}>
      <Link className="flex items-center text-3xl md:text-4xl py-4 px-2" to="/">
        <FontAwesomeIcon icon={faPiedPiperSquare} className="mr-2 md:pb-2.5" />
        <div className="flex flex-row justify-start content-center pt-2.5 md:pt-0">
          <div className="font-bold">TENNS</div>
          <div className="font-light">CENTRL</div>
        </div>
      </Link>
    </div>
  );
}
CompanyLogo.propTypes = {
  clickHandler: PropTypes.func.isRequired
};
