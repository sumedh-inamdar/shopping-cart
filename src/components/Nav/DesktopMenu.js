import React from 'react';
import { Link } from 'react-router-dom';

export default function DesktopMenu() {
  return (
    <div className="hidden md:flex items-center space-x-1">
      <Link
        className="py-4 px-2 navLink text-2xl font-light hover:text-zinc-600 transition duration-300"
        to="/">
        HOME
      </Link>
      <Link
        className="py-4 px-2 navLink text-2xl font-light hover:text-zinc-600 transition duration-300"
        to="/store">
        STORE
      </Link>
      <Link
        className="py-4 px-2 navLink text-2xl font-light hover:text-zinc-600 transition duration-300"
        to="/about">
        ABOUT
      </Link>
    </div>
  );
}
