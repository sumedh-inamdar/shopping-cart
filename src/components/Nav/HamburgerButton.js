import React from 'react';
import PropTypes from 'prop-types';

export default function HamburgerButton({
  setMobileMenuVisible,
  mobileMenuVisible
}) {
  return (
    <div className="md:hidden flex items-center">
      <button
        className="outline-none mobile-menu-button"
        onClick={() => {
          setMobileMenuVisible(!mobileMenuVisible);
        }}>
        {mobileMenuVisible ? (
          <div className="text-4xl w-8 h-8 pt-1">X</div>
        ) : (
          <svg
            className="w-8 h-8"
            x-show="!showMenu"
            fill="none"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            stroke="currentColor">
            <path d="M4 6h16M4 12h16M4 18h16"></path>
          </svg>
        )}
      </button>
    </div>
  );
}
HamburgerButton.propTypes = {
  setMobileMenuVisible: PropTypes.func.isRequired,
  mobileMenuVisible: PropTypes.bool.isRequired
};
