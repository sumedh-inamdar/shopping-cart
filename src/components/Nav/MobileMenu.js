import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export default function MobileMenu({ clickHandler }) {
  return (
    <div className="transition duration-300" onClick={clickHandler}>
      <ul className="">
        <li className="active">
          <Link
            to="/"
            className="block text-sm px-4 py-4 text-white hover:bg-green-600 font-semibold">
            HOME
          </Link>
        </li>
        <li>
          <Link
            to="/store"
            className="block text-sm px-4 py-4 hover:bg-green-600 transition duration-300">
            STORE
          </Link>
        </li>
        <li>
          <Link
            to="/about"
            className="block text-sm px-4 py-4 hover:bg-green-600 transition duration-300">
            ABOUT
          </Link>
        </li>
      </ul>
    </div>
  );
}
MobileMenu.propTypes = {
  clickHandler: PropTypes.func.isRequired
};
