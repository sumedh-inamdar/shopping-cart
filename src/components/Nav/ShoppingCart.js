import React from 'react';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faShoppingCart } from '@fortawesome/free-solid-svg-icons';
import PropTypes from 'prop-types';

export default function ShoppingCart({ clickHandler, cartQty }) {
  return (
    <div className="relative flex items-center pt-2.5" onClick={clickHandler}>
      <Link
        className="text-2xl py-2 px-2 hover:text-zinc-600 transition duration-300"
        to="/checkout">
        <FontAwesomeIcon icon={faShoppingCart} />
        <div className="absolute top-[10px] left-[25px] bg-orange-600 rounded-[50%] h-6 w-6 m-auto">
          <div className="text-center w-full h-full text-base">{cartQty}</div>
        </div>
      </Link>
    </div>
  );
}
ShoppingCart.propTypes = {
  clickHandler: PropTypes.func.isRequired,
  cartQty: PropTypes.number.isRequired
};
