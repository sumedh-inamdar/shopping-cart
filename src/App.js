import { useState, useEffect } from 'react';
import { useLocation, BrowserRouter, Routes, Route } from 'react-router-dom';
import Nav from './components/Nav';
import Home from './components/Home';
import Store from './components/store/Store';
import About from './components/About';
import Checkout from './components/checkout/Checkout';
import Footer from './components/Footer';

function ScrollToTop(props) {
  const location = useLocation();
  useEffect(() => {
    window.scrollTo(0, 0);
  }, [location]);

  // eslint-disable-next-line react/prop-types
  return <>{props.children}</>;
}

function App() {
  const [cart, setCart] = useState([]);
  return (
    <div id="app" className="relative min-h-screen">
      <BrowserRouter>
        <div className="pb-[452px] md:pb-52">
          <Nav
            cartQty={cart.reduce(
              (previousValue, currentValue) =>
                previousValue + currentValue.quantity,
              0
            )}
          />
          <ScrollToTop>
            <Routes>
              <Route path="/" element={<Home />} />
              <Route
                path="/store/*"
                element={<Store cart={cart} setCart={setCart} />}></Route>
              <Route path="/about" element={<About />} />
              <Route
                path="/checkout"
                element={<Checkout cart={cart} setCart={setCart} />}
              />
            </Routes>
          </ScrollToTop>
        </div>
        <Footer />
      </BrowserRouter>
    </div>
  );
}

export default App;
