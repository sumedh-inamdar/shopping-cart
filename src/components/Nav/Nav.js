import React, { useEffect, useState } from 'react';
import MobileMenu from './MobileMenu';
import ShoppingCart from './ShoppingCart';
import DesktopMenu from './DesktopMenu';
import HamburgerButton from './HamburgerButton';
import CompanyLogo from './CompanyLogo';

export default function Nav() {
  const [mobileMenuVisible, setMobileMenuVisible] = useState(false);

  useEffect(() => {
    if (mobileMenuVisible)
      window.addEventListener('click', mobileMenuClickHandler);
    return () => {
      window.removeEventListener('click', mobileMenuClickHandler);
    };
  }, [mobileMenuVisible]);

  function mobileMenuClickHandler(event) {
    if (event.target.closest('nav') === null) closeMobileMenu();
  }

  function closeMobileMenu() {
    setMobileMenuVisible(false);
    window.removeEventListener('click', mobileMenuClickHandler);
  }

  return (
    <nav className="bg-gradient-to-r from-teal-500 to-indigo-500 text-white">
      <div className="max-w-6xl mx-auto px-6">
        <div className="flex justify-between">
          <HamburgerButton
            mobileMenuVisible={mobileMenuVisible}
            setMobileMenuVisible={setMobileMenuVisible}
          />
          <div className="flex space-x-7 min-w-fit">
            <CompanyLogo clickHandler={closeMobileMenu} />
            <DesktopMenu />
          </div>
          <ShoppingCart clickHandler={closeMobileMenu} />
        </div>
      </div>
      {mobileMenuVisible && <MobileMenu clickHandler={closeMobileMenu} />}
    </nav>
  );
}
