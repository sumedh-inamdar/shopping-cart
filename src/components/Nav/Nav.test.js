import '@testing-library/jest-dom';
import { getByText, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import { CartContext } from '../../App';
import Nav from './Nav';

const resizeWindow = (x, y) => {
  window.innerWidth = x;
  window.innerHeight = y;
  window.dispatchEvent(new Event('resize'));
};

describe('Nav component', () => {
  const cart = [];

  it('snapshot check', () => {
    const { container } = render(
      <CartContext.Provider value={cart}>
        <Nav />
      </CartContext.Provider>,
      { wrapper: BrowserRouter }
    );
    expect(container).toMatchSnapshot();
  });

  it('mobile menu render', () => {
    resizeWindow(390, 844);
    render(
      <CartContext.Provider value={cart}>
        <Nav />
      </CartContext.Provider>,
      { wrapper: BrowserRouter }
    );
    expect(screen.queryByRole('link', { name: 'home-mobile-menu' })).toBeNull();

    const hamburger = screen.getByRole('button', { name: /hamburger/i });
    userEvent.click(hamburger);

    expect(
      screen.getByRole('link', { name: 'home-mobile-menu' })
    ).toBeVisible();
  });

  it('shopping cart quantity 0', () => {
    render(
      <CartContext.Provider value={cart}>
        <Nav />
      </CartContext.Provider>,
      { wrapper: BrowserRouter }
    );
    const shoppingCart = screen.getByRole('link', { name: 'shopping-cart' });
    expect(getByText(shoppingCart, 0)).toBeInTheDocument();
  });

  it('shopping cart quantity 5', () => {
    cart.push({ quantity: 2 }, { quantity: 3 });
    render(
      <CartContext.Provider value={cart}>
        <Nav />
      </CartContext.Provider>,
      { wrapper: BrowserRouter }
    );
    const shoppingCart = screen.getByRole('link', { name: 'shopping-cart' });
    expect(getByText(shoppingCart, 5)).toBeInTheDocument();
  });
});
