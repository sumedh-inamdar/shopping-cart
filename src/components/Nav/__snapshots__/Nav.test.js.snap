// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component snapshot check 1`] = `
<div>
  <nav
    class="bg-gradient-to-r from-teal-500 to-indigo-500 text-white"
  >
    <div
      class="max-w-6xl mx-auto px-6"
    >
      <div
        class="flex justify-between"
      >
        <div
          class="md:hidden flex items-center"
        >
          <button
            aria-label="Hamburger"
            class="outline-none mobile-menu-button"
          >
            <svg
              class="w-8 h-8"
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              x-show="!showMenu"
            >
              <path
                d="M4 6h16M4 12h16M4 18h16"
              />
            </svg>
          </button>
        </div>
        <div
          class="flex space-x-7 min-w-fit"
        >
          <div>
            <a
              class="flex items-center text-2xl md:text-4xl py-4 px-2"
              href="/"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-pied-piper-square mr-2 md:pb-2.5"
                data-icon="pied-piper-square"
                data-prefix="fab"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M32 419L0 479.2l.8-328C.8 85.3 54 32 120 32h327.2c-93 28.9-189.9 94.2-253.9 168.6C122.7 282 82.6 338 32 419M448 32S305.2 98.8 261.6 199.1c-23.2 53.6-28.9 118.1-71 158.6-28.9 27.8-69.8 38.2-105.3 56.3-23.2 12-66.4 40.5-84.9 66h328.4c66 0 119.3-53.3 119.3-119.2-.1 0-.1-328.8-.1-328.8z"
                  fill="currentColor"
                />
              </svg>
              <div
                class="flex flex-row justify-start content-center pt-2.5 md:pt-0"
              >
                <div
                  class="font-bold"
                >
                  TENNS
                </div>
                <div
                  class="font-light"
                >
                  CENTRL
                </div>
              </div>
            </a>
          </div>
          <div
            class="hidden md:flex items-center space-x-1"
          >
            <a
              class="py-4 px-2 navLink text-2xl font-light hover:text-zinc-600 transition duration-300"
              href="/"
            >
              HOME
            </a>
            <a
              class="py-4 px-2 navLink text-2xl font-light hover:text-zinc-600 transition duration-300"
              href="/store"
            >
              STORE
            </a>
            <a
              class="py-4 px-2 navLink text-2xl font-light hover:text-zinc-600 transition duration-300"
              href="/about"
            >
              ABOUT
            </a>
          </div>
        </div>
        <div
          class="relative flex items-center pt-2.5"
        >
          <a
            aria-label="shopping-cart"
            class="text-2xl py-2 px-2 hover:text-zinc-600 transition duration-300"
            href="/checkout"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-cart-shopping "
              data-icon="cart-shopping"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M96 0C107.5 0 117.4 8.19 119.6 19.51L121.1 32H541.8C562.1 32 578.3 52.25 572.6 72.66L518.6 264.7C514.7 278.5 502.1 288 487.8 288H170.7L179.9 336H488C501.3 336 512 346.7 512 360C512 373.3 501.3 384 488 384H159.1C148.5 384 138.6 375.8 136.4 364.5L76.14 48H24C10.75 48 0 37.25 0 24C0 10.75 10.75 0 24 0H96zM128 464C128 437.5 149.5 416 176 416C202.5 416 224 437.5 224 464C224 490.5 202.5 512 176 512C149.5 512 128 490.5 128 464zM512 464C512 490.5 490.5 512 464 512C437.5 512 416 490.5 416 464C416 437.5 437.5 416 464 416C490.5 416 512 437.5 512 464z"
                fill="currentColor"
              />
            </svg>
            <div
              class="absolute top-[10px] left-[25px] bg-orange-600 rounded-[50%] h-6 w-6 m-auto"
            >
              <div
                class="text-center w-full h-full text-base"
              >
                0
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
